src/coreclr/ToolBox/superpmi/superpmi-shim-collector/superpmi-shim-collector.cpp

```diff
@@ -142,6 +155,9 @@ extern "C"
 
 extern "C" DLLEXPORT void __stdcall jitStartup(ICorJitHost* host)
 {
+    // crossgen2 doesn't invoke DllMain, so optionally do initialization work here.
```

Comment:
I must admit I'm still somewhat confused here. IIRC, <code>LoadLibrary</code> (that's internally used by Crossgen2 to load the JIT) automatically calls <code>DllMain(DLL_PROCESS_ATTACH / DLL_THREAD_ATTACH)</code>. My understanding of the problem was that the clean shutdown doesn't happen - we don't unload the DLL upon finishing the compilation so that DllMain(DLL_PROCESS_DETACH / DLL_THREAD_DETACH) doesn't happen and doesn't end up flushing some files or statistic logs or whatnot. I don't have precise understanding of what happens in <code>InitializeShim</code> but I would be very suprised if DllMain(DLL_PROCESS_ATTACH / DLL_THREAD_ATTACH) wasn't run, that would nullify all known LoadLibrary[Ex] invariants I'm aware of.

Reply:
The issue is only on Linux using the PAL. On Windows, it works fine. I should clarify the comment.

Reply:
I see, thanks for clarifying!